### DISCUSSION TITLE
Feature Request: Add support for authenticated API requests in DuckDB-wasm

### DISCUSSION THREAD

**Discussion by nite** [duckdb/duckdb#13351] [Ideas]
Currently, DuckDB-wasm doesn't support loading data directly from REST APIs that require authentication headers. This limits its usefulness in browser environments where data needs to be fetched from authenticated sources.

Proposed Solution:
Extend the HTTP functionality in DuckDB-wasm to support custom headers, particularly authentication headers. This would allow users to directly query authenticated APIs without needing to use intermediate JavaScript to fetch and load the data.

Use Case:
This would be particularly useful for web applications that need to query data from authenticated APIs directly within DuckDB-wasm, improving performance and reducing code complexity.

Additional Context:
The current workaround involves fetching data via JavaScript and then loading it into DuckDB-wasm, which is less efficient and more complex to implement.

**Comment by cfahlgren1** [comment]
I think the latest DuckDB 1.1.0 should be able to work for your use case. See [DuckDB Shell](https://shell.duckdb.org/#queries=v0,CREATE-SECRET-http3-(%0A----TYPE-HTTP%2C-%0A----EXTRA_HTTP_HEADERS-MAP%7B%0A%09%09'Authorization'%3A-'Bearer-%24%7Bmy-token%7D'%2C%0A%09%09'my_own_header'%3A-'my_special_value'%0A%09%7D%0A)~%0Afrom-'https%3A%2F%2Fmah%20server%2Fsome%20file.parquet'%0A~) 

[PR](https://github.com/duckdb/duckdb/pull/13368)

**Reply by nite** [reply]
Great stuff - for reference https://duckdbsnippets.com/snippets/184/query-an-authenticated-api-endpoint

**Reply by nite** [reply]
Should this work in duckdb-wasm yet? It works on command line for us, but not in wasm - doesnt seem to be adding headers and we get ```Invalid Error: Opening file '...' failed with error: NetworkError: Failed to execute 'send' on 'XMLHttpRequest': Failed to load```

**Reply by cfahlgren1** [reply]
No I don't think its supported yet. Probably needs an issue at https://github.com/duckdb/duckdb-wasm. I thought there was one, but couldn't find.

**Reply by nite** [reply]
Ok, thanks - added one now https://github.com/duckdb/duckdb-wasm/issues/1967!

**Reply by nite** [reply]
Hi @cfahlgren1 - is there anyone you know of who could us get this in? Happy to collab on it.
